### DISCUSSION TITLE
Chaging the style of a button at runtime didn't work

### DISCUSSION THREAD

**Discussion by jeff9571** [AvaloniaUI/Avalonia#10698] [Q&A]
I want to change the style about the pointerover background color of a button, like when pointer over the button,it's background can be blue or red etc. But Only the first change to the background color when the mouse hovers over the control takes effect, subsequent changes do not work even though the Styles collection has been updated according to the debug information.  The code below: **(IBrush?)val)** will be changed at runtime
`  button.Styles.Remove(_customStyle);`  
 ` _customStyle = new Style(selector =>
                    selector.OfType<Button>().Class(":pointerover").Template().OfType<ContentPresenter>()
                        .Name("PART_ContentPresenter"))
                {
                    Setters = { new Setter(TemplatedControl.BackgroundProperty, (IBrush?)val) }
                };`  
`  button.Styles.Add(_customStyle);`
Ps. These code changes work for TextBlock, but they don't work for Button.

**Comment by timunie** [comment]
Why do do it that complicated? Please check out Styles in the docs and look up the original Button Styles.

docs: https://docs.avaloniaui.net/docs/styling
Button-Style (v0.10.18): https://github.com/AvaloniaUI/Avalonia/blob/401bc521d2947bf7699a530f3d6e43e7a32c6baf/src/Avalonia.Themes.Fluent/Controls/Button.xaml#L44-L49
